Transform PDF

A-Faris opened the pull request:
Create PDF summary report
closes #74

A-Faris committed:
added transform

Commit:
Update scores

fm1psy commented on pdf_report/transform_pdf.py:
```diff
@@ -0,0 +1,76 @@
+
+from extract_pdf import get_connection, query_db
+
+# [RealDictRow({'station_id': 7, 'waypoint_id': 8407, 'run_date': datetime.date(2024, 7, 22), 'booked_arrival': datetime.datetime(2024, 7, 22, 23, 35), 'actual_arrival': datetime.datetime(2024, 7, 22, 23, 34), 'booked_departure': datetime.datetime(2024, 7, 22, 23, 39), 'actual_departure': datetime.datetime(2024, 7, 22, 0, 1), 'service_id': 5097, 'station_crs': 'YRK', 'station_name': 'York Rail St ...
```
Was this comment a personal note? perhaps this note is no longer needed.
```suggestion
```

FarihaChoudhury replied:
ew why are there comments in a pull request? fix up

A-Faris replied:
removed it

A-Faris committed:
got data

A-Faris committed:
merging

Merge branch 'pdf_transform' of https://github.com/Zhi-704/Railway-Tracker into pdf_transform

Commit:
Update scores

fm1psy commented on pdf_report/extract_pdf.py:
```diff
@@ -28,16 +28,21 @@ def get_cursor(conn: connection) -> cursor:
     return conn.cursor(cursor_factory=RealDictCursor)
 
 
+def query_db(conn: connection, query: str) -> list[tuple]:
```
I like the use of a function to run queries, helps reduce dry code

A-Faris replied:
thank you

Zhi-704 commented on .util/pylint_scores.txt:
```diff
@@ -29,6 +29,14 @@ Your code has been rated at 10.00/10
 Your code has been rated at 10.00/10
 
+************* Module transform_pdf
+pdf_report/transform_pdf.py:40:0: C0301: Line too long (122/100) (line-too-long)
+pdf_report/transform_pdf.py:41:0: C0301: Line too long (127/100) (line-too-long)
+pdf_report/transform_pdf.py:76:0: C0116: Missing function or method docstring (missing-function-docstring)
+
+-----------------------------------
+Your code has been rated at 8.93/10
```
Smh has to be above 9

A-Faris replied:
Done

fm1psy reviewed (dismissed):
This looks ready to merge

A-Faris committed:
pdf

A-Faris committed:
Merge branch 'main' into pdf_transform

Commit:
Update scores

A-Faris committed:
created group bar pdf

A-Faris committed:
merging
Merge branch 'pdf_transform' of https://github.com/Zhi-704/Railway-Tracker into pdf_transform

Commit:
Update scores

A-Faris committed:
pdf has station crs

A-Faris committed:
merging
Merge branch 'pdf_transform' of https://github.com/Zhi-704/Railway-Tracker into pdf_transform

Commit:
Update scores

A-Faris committed:
Merge branch 'main' into pdf_transform

Commit:
Update scores

A-Faris committed:
finished pdf

A-Faris committed:
merging
Merge branch 'pdf_transform' of https://github.com/Zhi-704/Railway-Tracker into pdf_transform

Commit:
Update scores

A-Faris committed:
file structure

A-Faris committed:
merging
Merge branch 'pdf_transform' of https://github.com/Zhi-704/Railway-Tracker into pdf_transform

FarihaChoudhury commented on pdf_report/transform_pdf.py:
```diff
@@ -0,0 +1,255 @@
+    return pisa_status.err
+
+
+def transform_pdf() -> None:
```
I think this function as a whole is quite big, maybe you could separate it into smaller functions by similarity

FarihaChoudhury commented on pdf_report/transform_pdf.py:
```diff
@@ -0,0 +1,255 @@
+              - {row['station_name']}</li>"
+    station_info_html += "</ul>"
+
+    html_report = f"""
```
could have a function that just returns this html report? to reduce the clutter in ```transform_pdf.py```

A-Faris replied:
Done

A-Faris committed:
pylinting

Commit:
Update scores

A-Faris committed:
pyliniting

A-Faris committed:
merging
Merge branch 'pdf_transform' of https://github.com/Zhi-704/Railway-Tracker into pdf_transform

Commit:
Update scores

A-Faris committed:
fixed axis label

A-Faris committed:
merging
Merge branch 'pdf_transform' of https://github.com/Zhi-704/Railway-Tracker into pdf_transform

Zhi-704 committed:
Merge branch 'main' into pdf_transform

Commit:
Update scores

A-Faris committed:
more functions

Commit:
Update scores

FarihaChoudhury commented on pdf_report/transform_pdf.py:
```diff
@@ -0,0 +1,273 @@
+        ['avg_arrive_delay_long_minutes', 'avg_departure_delay_long_minutes'],
+        'Average Delay more than 1 min by Station', 'Average Delay more than 1 min (minutes)')
+
+    html_report = generate_html_report(delayed_df, cancellation_chart,
```
looks so much better

A-Faris committed:
reorder

A-Faris committed:
merging
Merge branch 'pdf_transform' of https://github.com/Zhi-704/Railway-Tracker into pdf_transform

FarihaChoudhury reviewed (dismissed):
Good work :) love the report

Commit:
Update scores

FarihaChoudhury reviewed (approved):
amazin
